### ISSUE TITLE
osxfuse not compatible with MacOS Mojave

### ISSUE THREAD

**Issue by gvansanden** [osxfuse/osxfuse#542]
I just upgraded and get the message that FUSE is too old for my OS.
I'm on 3.7.1 and update check gives no newer (beta) versions

**Comment by gvansanden** [comment]
I just noticed a 3.8 version on the site, but the auto-updater does not pick it up (with or without beta selected).  Fuse works now

**Comment by dongepulango** [comment]
Same here, FUSE is not working anymore after updating to macOS Mojave . I'm in v3.8.2

**Comment by ChrisCarrick1728** [comment]
It is also not working for me after I updated to Mojave. I'm on v3.7.1

**Comment by bfleischer** [comment]
Chris, you need to update to version 3.8.2. 3.7.1 does not Support nacOS Mojave.

**Comment by ChrisCarrick1728** [comment]
Updating to 3.8.2 worked for me. I had to download and install the .dmg as the update tool in the system preferences pane did not work. Thanks for the quick reply bfleischer!

**Comment by pabloberrios** [comment]
I have a Macbook Pro (13-inch, 2017) with Mojave 10.14 and not work osxfuse v3.8.2. The NTFS Disks still appear read only. 
Anyone have some tips?

**Comment by eduardsui** [comment]
Same problem with Mojave. Using version 3.8.2, tried also to reinstall, still not working.

**Comment by eduardsui** [comment]
It works now, it was a race condition in my code.

**Comment by eduardsui** [comment]
If it helps with debugging this, I've noticed that on Mojave, `getattr(/)` gets called when mounting, so this code:
```
    if ((ch = fuse_mount(mountpoint, &args)) != NULL) {
        se = fuse_new(ch, &args, &edfs_fuse, sizeof(edfs_fuse), NULL);
        // here I initialize my network thread
        edfs_edwork_init(edfs_context, port);
...
    }
```
This doesn't work anymore (before Mojave, when using libfuse or dokany this worked). Now, `fuse_mount` results in a query for `getattr("/")`, which returns, in my case, ENOENT (because the network thread is not yet initialized). I've adjusted the code by ensuring the network thread is initialized before fuse_mount gets called. Also, this call gets cached, so event after the network thread gets initialized, this continues to return ENOENT (this is a good thing).

**Comment by cristianmenghi** [comment]
Same problem, installing from brew cask v 3.8.2 can't mount NTFS

**Comment by xmfbit** [comment]
Update to 3.8.2 just solved my problem.

**Comment by ITguyAtlanta** [comment]
Same issue. 
Upgraded to Mojave 10.14, upgraded Fuse to 3.8.2
Getting failure to mount error
Says this edition of Fuse is too old for the OS
Any help appreciated.

**Comment by asapkov** [comment]
Unable to upgrade to 3.8.2. Downloaded dmg installer from the site. Everything is fine until "Click install to perform a standard installation blah-blah" message. Clicking on "install" does nothing. No errors or crashes, but no installation as well.

**Comment by ghost** [comment]
@asapkov, you could see errors and messages in `/var/log/install.log` with **Console** application.

**Comment by mkexc** [comment]
Not working for me also.

**Comment by caozilu** [comment]
works as read-only, unable to do write actions

**Comment by bfleischer** [comment]
Everyone with NTFS issues, please read https://github.com/osxfuse/osxfuse/wiki/NTFS-3G. You need NTFS-3G and FUSE for be able to write on NTFS volumes. FUSE for macOS 3.8.2 works fine on macOS Mojave.

Please know the comments like "Same problem" or "Same issue" are not very helpful. Be more specific. Which file system (or applications) are you trying to use?

**Comment by iw5ejm** [comment]
MacOS Mojave 10.14.1 (18B75) not working: no network device displayed, fuse 3.8.2 correctly installed

**Comment by LBensman** [comment]
Same issue as reported by @iw5ejm .

Path:
1. Upgraded to Mojave and fuse stopped working.
2. Upgraded FUSE to 3.8.2 and it started working.
3. Today macos installed update which made it now macos 10.14.1.
4. FUSE now refuses to work - fails to mount remote linux drive.
```
$ sshfs -V
SSHFS version 2.10
OSXFUSE 3.8.2
FUSE library version: 2.9.7
```
Note that https://osxfuse.github.io/ lists sshfs stable release as 2.5.0.  Seems I have more recent one, so I assume I don't need to run that pkg?

**Comment by keeninter** [comment]
OpenDrive (www.opendrive.com) for Mac does not work on Mojave with osxfuse 3.8.2 on some machines: mounting is done but then only read access is enabled, write operations are not performed.

**Comment by daynaskully** [comment]
Updated a bunch of mac's to mojave now that it's been out for a few months (developer preview of course being available a few months before that).

Cant get fuse to work (version 3.7 doesn't offer updates) manually installing the most recent version 3.8 doesn't work either -- same problem of "too old for os".

Is this project abandoned for osx or something? Any alternatives?

**Comment by misku** [comment]
@daynaskully The project is not abandoned, but it's developed *for free* during @bfleischer *free time*.

**Comment by senochow** [comment]
https://osxfuse.github.io/
3.8.2 is ok

**Comment by esetleges** [comment]
I guess (if you have done it before) you have to disable the SIP in recovery mode then move the ntfs-3g tools into the system (root) and after enable SIP again.

I got no right to write NTFS (after upgrading to Mojave) while was not repeating these steps until the last point (csrutils disable, blabla, csrutils enable):
https://www.howtogeek.com/236055/how-to-write-to-ntfs-drives-on-a-mac/

Advise that to update (upgrade) Xcode first, then Homebrew, FUSE, reinstall NTFS-3g and afterwards, repeat that last step (if you have done it before)

Now it is working fine again, I can write my external disk.

Mac OS 10.14.2/FUSE 3.8.3/NTFS-3G 2017.3.23

**Comment by akositey** [comment]
> Advise that to update (upgrade) Xcode first, then Homebrew, FUSE, reinstall NTFS-3g and afterwards, repeat that last step (if you have done it before)

Thanks you for this! I'll share mine for others to try.

In my case also updating Fuse did not do the job since. I think after update(maybe to 10.14?) somehow Homebrew went missing in the system but FUSE remained. I did these steps:
1. Update Xcode from App Store (Since it prompted me to do so the other day)
2. Update Fuse to latest 3.8.3
3. Install Homebrew
4. Reinstalled NTFS-3g (followed the [NTFS-3G wiki](https://github.com/osxfuse/osxfuse/wiki/NTFS-3G) for Auto-mounting, probably because after update the mount_ntfs, which was replaced, was erased... so I did the steps again.)

Hope it helps somebody too. :)
